ee/localserver/request-query.go

```diff
@@ -19,48 +21,89 @@ func (ls *localServer) requestQueryHanlderFunc(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	var body map[string]any
+	var body map[string]string
 	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
 		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
 		return
 	}
 
-	queryRaw, ok := body["query"]
-	if !ok {
+	query, ok := body["query"]
+	if !ok || query == "" {
 		sendClientError(w, "no query key found in request body json")
 		return
 	}
 
-	query, ok := queryRaw.(string)
-	if !ok {
-		sendClientError(w, fmt.Sprintf("query value not a string: %s", query))
+	var results []map[string]string
+	var err error
+	backoff.WaitFor(func() error {
+		results, err = ls.querier.Query(query)
+		return err
+	}, 1*time.Second, 250*time.Millisecond)
+
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error executing query: %s", err))
+		return
+	}
+
+	jsonBytes, err := json.Marshal(results)
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error marshalling results to json: %s", err))
+		return
+	}
+
+	w.Write(jsonBytes)
+}
+func (ls *localServer) requestRunScheduledQueryHandler() http.Handler {
+	return http.HandlerFunc(ls.requestRunScheduledQueryHanlderFunc)
+}
+
+func (ls *localServer) requestRunScheduledQueryHanlderFunc(w http.ResponseWriter, r *http.Request) {
+	if r.Body == nil {
+		sendClientError(w, "request body is nil")
+		return
+	}
+
+	var body map[string]string
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
 		return
 	}
 
-	if query == "" {
-		sendClientError(w, "empty query")
+	name, ok := body["name"]
+	if !ok || name == "" {
+		sendClientError(w, "no name key found in request body json")
 		return
 	}
 
+	scheduledQueryQuery := fmt.Sprintf("select query from osquery_schedule where name = '%s'", name)
```

Comment:
Making sure I'm understanding right -- this is how it works?

1. K2 calls `/runscheduledquery` with the name of the scheduled query
1. Launcher returns the query
1. K2 calls `/query` with the query returned in the previous step to actually run it

Reply:
So the driver behind this is that the JS bridge has to use GET requests passing the query (in a nacl box) as a URL parameter. This means there is a limit on the size of the query. This endpoint is intended to be a work around for that. It ought to work like this:

1. K2 looks up the name of a osquery scheduled query it want to be run. The same query should be available to launcher in the osquery_schedule table.
2. K2 calls `/scheduledquery`, with body `{ "name": "name_of_scheduled_query" }`
3. Launcher looks up the query's sql using the provided name
4. Launcher executes the query
5. Launcher returns results

I'll add a comment ents to the func about this.

Reply:
Ah ok, I just completely missed where it was calling `requestQueryHanlderFunc` at the end of this func and that's how the query was getting run! Thanks for adding the comments